org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/id/IdCodec.java

```diff
@@ -99,19 +155,60 @@ public String toUnqualified(IId id) {
       if (mapper == null) {
         throw new PlatformException("Missing raw type mapper for wrapped type {}, id type {}", value.getClass(), id.getClass());
       }
-      return mapper.apply(value);
+      return encryptUnqualified(id.getClass(), mapper.apply(value), flags);
     }
     else if (id instanceof ICompositeId) {
       List<? extends IId> components = ((ICompositeId) id).unwrap();
-      return components.stream()
-          .map(this::toUnqualified)
+      Set<IIdCodecFlag> flagsWithoutEncryption = flags.stream()
```

Comment:
add javadoc indicating that encryption flag is removed since composite id is signed as complete id and not part by part